src/pubsub.js

```diff
@@ -0,0 +1,573 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ['error', 8] */
+'use strict'
+
+const expect = require('chai').expect
+const series = require('async/series')
+const waterfall = require('async/waterfall')
+const parallel = require('async/parallel')
+const whilst = require('async/whilst')
+const each = require('async/each')
+
+function waitForPeers (ipfs, topic, peersToWait, callback) {
+  const i = setInterval(() => {
+    ipfs.pubsub.peers(topic, (err, peers) => {
+      if (err) {
+        return callback(err)
+      }
+
+      const missingPeers = peersToWait
+            .map((e) => peers.includes(e))
+            .filter((e) => !e)
+
+      if (missingPeers.length === 0) {
+        clearInterval(i)
+        callback()
+      }
+    })
+  }, 500)
+}
+
+function spawnWithId (factory, callback) {
+  waterfall([
+    (cb) => factory.spawnNode(cb),
+    (node, cb) => node.id((err, res) => {
+      if (err) {
+        return cb(err)
+      }
+      node.peerId = res
+      cb(null, node)
+    })
+  ], callback)
+}
+
+function makeCheck (n, done) {
+  let i = 0
+  return (err) => {
+    if (err) {
+      return done(err)
+    }
+
+    if (++i === n) {
+      done()
+    }
+  }
+}
```

Comment:
I honestly found this to be simpler to use and understand. But not opposed to switching.